### ISSUE TITLE
Wha's the preferred way of installing modules ?

### ISSUE THREAD

**Issue by who** [rroemhild/docker-ejabberd#84]
Using this images, what's the preferred way of installing modules into the ejabberd sources folder?

Currently, I am doing this:

1. docker run ... rroemhild/ejabberd
2. *wait for ejabberd to start ...*
3. docker cp ./mod_my_custom_module ejabberd:/opt/ejabberd/.ejabberd-modules/sources/.
4. docker exec -it ejabberd ejabberdctl module_install mod_my_custom_module
5. docker exec -it ejabberd ejabberdctl restart

Can you think of a good way to install these modules before ejabberd starts initially?

**Comment by who** [comment]
@rroemhild 

@rroemhild 

Here's a solution I have experimented with:

```
    docker run -d \
        --name "ejabberd" \
        -p 5222:5222 \
        -p 5269:5269 \
        -p 5280:5280 \
        -h 'xmpp.example.de' \
        -e "XMPP_DOMAIN=example.de" \
        -e "ERLANG_NODE=ejabberd" \
        -e "EJABBERD_ADMINS=admin@example.de admin2@example.de" \
        -e "EJABBERD_USERS=admin@example.de:password1234 admin2@example.de" \
        -e "TZ=Europe/Berlin" \
        -e "EJABBERD_SOURCE_MODULES=mod_hello_world mod_foo_bar" \
        -v /tmp/modules_to_install:/opt/ejabberd/module_source \
        rroemhild/ejabberd
```

Note the new environment variable above, `EJABBERD_SOURCE_MODULES`. The value of this variable are module names, separated by space. This allows a post processing script to iterate over `$EJABBERD_SOURCE_MODULES`. For each of the module names found inside `$EJABBERD_SOURCE_MODULES`, the post processing script can attempt to copy the module folder from the folder `$EJABBERD_HOME/module_source/` into the ejabberd modules folder. After the copy occurs, `ejabberdctl module_check $module_name` can be performed, and if that succeeds, `$ejabberdctl module_install $module_name` can be performed.

**tldr;**
A new environment variable (`EJABBERD_SOURCE_MODULES`) defines modules which should be copied from a shared volume and installed into ejabberd.

**Comment by who** [comment]
I've been testing the approach above and found it to be working very well. I will submit a PR with the post script.
